docs/gallery_code/general/plot_projections_and_annotations.py

```diff
@@ -70,8 +70,15 @@ def make_plot(projection_name, projection_crs):
     margin_fractions = np.array([margin, 1.0 - margin])
     x_lower, x_upper = x_start + (x_end - x_start) * margin_fractions
     y_lower, y_upper = y_start + (y_end - y_start) * margin_fractions
-    box_x_points = x_lower + (x_upper - x_lower) * np.array([0, 1, 1, 0, 0])
-    box_y_points = y_lower + (y_upper - y_lower) * np.array([0, 0, 1, 1, 0])
+    steps = np.linspace(0, 1)
+    zeros, ones = np.zeros(steps.size), np.ones(steps.size)
+    x_delta, y_delta = (x_upper - x_lower), (y_upper - y_lower)
+    x_points = x_lower + x_delta * np.concatenate(
+        (steps, ones, steps[::-1], zeros)
+    )
+    y_points = y_lower + y_delta * np.concatenate(
+        (zeros, steps, ones, steps[::-1])
+    )
```

Comment:
`cartopy` should really just be better at this... it's always been pretty poor at projecting a line with an appropriate level of intermediate line point interpolation. Clearly something has changed between 0.19 and 0.20, so for this gallery example I've opted to take matters into our own hands... which isn't ideal, but perhaps we can eventually make `cartopy` better at this.

Here is this gallery image using `cartopy` 0.19 (without the above change)
![image](https://user-images.githubusercontent.com/2051656/134135402-b150ea3c-5c79-4e67-b189-415f071e2a9e.png)

Here is this gallery image using `cartopy` 0.20 (without the above change)
![image](https://user-images.githubusercontent.com/2051656/134135808-5a6e9eb6-d799-4893-a560-31adcb8af2b2.png)

Here is this gallery image using `cartopy` 0.20 (**with** the above change)
![image](https://user-images.githubusercontent.com/2051656/134136184-7825da6c-533b-4401-a4ff-45e6a65a12a3.png)

Reply:
What is the issue I am looking for here? The jagged-ness of the white dotty lines in the version without the explicit steps?

Reply:
@jamesp Yes, the inset white dashed margin line that over plots the pcolormesh (the middle group of plots) fails the imagehash due to the fact that `cartopy` isn't really projecting the line with sufficient intermediate interpolated points. 

Hard to see, but the second plot from the middle group of plots has the northern margin line cutting straight across and goes outside the pcolormesh boundary (hard to explain in words, sorry)

Reply:
Even with the original plots (first group) you can see that the curves are made up of straight lines